internal/cortex/querier/queryrange/stats_middleware.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) The Cortex Authors.
+// Licensed under the Apache License 2.0.
+
+package queryrange
+
+import (
+	"context"
+
+	"github.com/thanos-io/thanos/internal/cortex/querier/stats"
+)
+
+type statsMiddleware struct {
+	next       Handler
+	forceStats bool
+}
+
+func NewStatsMiddleware(forceStats bool) Middleware {
+	return MiddlewareFunc(func(next Handler) Handler {
+		return statsMiddleware{
+			next:       next,
+			forceStats: forceStats,
+		}
+	})
+}
+
+func (s statsMiddleware) Do(ctx context.Context, r Request) (Response, error) {
+	if s.forceStats {
+		r = r.WithStats("all")
+	}
```

Comment:
I don't fully understand why we need a middleware here. Why we cannot do that in query frontend `ServeHTTP` function?

Reply:
Yep, you are right. That is much simpler, will do that.

Reply:
we will have to keep the middleware though, because only here we have access to the decoded response of the upstream queriers. I will move this logic to the handler though.